### PR TITLE
[NUI] Add GridRecycleLayotManager

### PR DESCRIPTION
**Pull request by neostom432**
### Description of Change ###
Grid Layout for RecyclerView.

When user set Orientation, GridRecycleLayoutManager will lay out items using only one factor between Row and Column.

Horizontal => Row
Vertical => Column

### API Changes ###
NONE

### CODE REVIEW

File: src/Tizen.NUI.Components/Controls/RecyclerView/GridRecycleLayoutManager.cs [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+
+        private int row = 1;
+
+        /// <summary>
+        /// [draft ]Get/Set the number of rows in the grid
+        /// </summary>
+        /// <since_tizen> 8 </since_tizen>
+        // This will be public opened after ACR done. (Before ACR, need to be hidden as Inhouse API)
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public int Row
```
Comment (dasg34) [inline src/Tizen.NUI.Components/Controls/RecyclerView/GridRecycleLayoutManager.cs:41]: Generally, `Row` is used as a word to represent specific row index of child.
I recommend `Rows` or `RowCount`.

File: src/Tizen.NUI.Components/Controls/RecyclerView/GridRecycleLayoutManager.cs [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+        private int column = 1;
+
+
+        /// <summary>
+        /// [Draft] Get/Set the number of columns in the grid
+        /// </summary>
+        /// <since_tizen> 8 </since_tizen>
+        // This will be public opened after ACR done. (Before ACR, need to be hidden as Inhouse API)
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public int Column
```
Comment (dasg34) [inline src/Tizen.NUI.Components/Controls/RecyclerView/GridRecycleLayoutManager.cs:67]: Generally, `Column` is used as a word to represent specific column index of child.
I recommend `Columns` or `ColumnCount`.

File: src/Tizen.NUI.Components/Controls/RecyclerView/GridRecycleLayoutManager.cs [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+                Math.Abs(scrollPosition) + (LayoutOrientation == Orientation.Horizontal ?
+                                                list.Size.Width : list.Size.Height)
+            );
+
+            float firstCheckPoint = LayoutOrientation == Orientation.Horizontal ? item.Position.X : item.Position.Y;
+            float secondCheckPoint = LayoutOrientation == Orientation.Horizontal ?
+                                        firstCheckPoint + item.Size.Width :
+                                        firstCheckPoint + item.Size.Height;
+
+            // Tizen.Log.Error("NUI", "[1p] "+visibleArea.X+ " =< "+firstCheckPoint+" =< "+visibleArea.Y+" ==== \n");
```
Comment (dasg34) [inline src/Tizen.NUI.Components/Controls/RecyclerView/GridRecycleLayoutManager.cs:102]: We need to leave commented logs?

**Comment by neostom432** [comment]
Abandon because it keeps asking resolve conflicts but there is no conflict.
https://github.com/Samsung/TizenFX/pull/1685

new pull request

**Pull request closed without merging**